### PR TITLE
Set default browser improvements android

### PR DESCRIPTION
**Pull request by tapanmodh**
Resolves: https://github.com/brave/internal/issues/837

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/brave_java_sources.gni [hunk 1]
```diff
@@ -344,7 +346,7 @@ if (enable_brave_referrals) {
   brave_java_sources += [ "../../brave/android/java/org/chromium/chrome/browser/util/BraveReferrer.java" ]
 }
 
-brave_browser_ui_android_appmenu_internal_sources = [ "//brave/browser/ui/android/appmenu/internal/java/src/org/chromium/chrome/browser/ui/appmenu/BraveAppMenu.java" ]
+brave_browser_ui_android_appmenu_internal_sources = [ "//brave/browser/ui/android/appmenu/internal/java/src/org/chromium/chrome/browser/ui/appmenu/BraveAppMenu.java", "//brave/browser/ui/android/default_browser_promo/java/src/org/chromium/chrome/browser/ui/default_browser_promo/BraveDefaultBrowserPromoUtils.java"]
```
Comment (samartnik) [inline android/brave_java_sources.gni:349]: I believe lint won't like it in 1 line

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java [hunk 2]
```diff
@@ -562,15 +573,21 @@ public void finishNativeInitialization() {
             BraveRewardsHelper.setNextRewardsOnboardingModalDate(calender.getTimeInMillis());
         }
 
-        if (SharedPreferencesManager.getInstance().readInt(BravePreferenceKeys.BRAVE_APP_OPEN_COUNT)
+        /*if
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/app/BraveActivity.java:576]: Are we going to use commented code going forward? If not I'd just remove it.

File: android/java/org/chromium/chrome/browser/notifications/retention/RetentionNotificationUtil.java [hunk 3]
```diff
@@ -95,7 +99,7 @@ public static RetentionNotification getNotificationObject(String notificationTyp
         return mNotificationMap.get(notificationType);
     }
 
-    @SuppressLint("NotificationTrampoline")
+    //@SuppressLint("NotificationTrampoline")
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/notifications/retention/RetentionNotificationUtil.java:102]: Just remove it if lint is fixed.

File: android/java/org/chromium/chrome/browser/set_default_browser/BraveSetDefaultBrowserUtils.java [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    public static void setBraveDefaultDontAsk() {
+        SharedPreferencesManager.getInstance().writeBoolean(
+                BravePreferenceKeys.BRAVE_DEFAULT_DONT_ASK, true);
+    }
+
+    public static boolean isBraveDefaultDontAsk() {
+        return SharedPreferencesManager.getInstance().readBoolean(
+                BravePreferenceKeys.BRAVE_DEFAULT_DONT_ASK, false);
+    }
+}
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/set_default_browser/BraveSetDefaultBrowserUtils.java:234]: nit: new line

File: android/java/org/chromium/chrome/browser/set_default_browser/OnBraveSetDefaultBrowserListener.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.chromium.chrome.browser.set_default_browser;
+
+public interface OnBraveSetDefaultBrowserListener {
+    default public void OnCheckDefaultResume(){};
+}
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/set_default_browser/OnBraveSetDefaultBrowserListener.java:12]: nit: new line

File: android/java/org/chromium/chrome/browser/set_default_browser/SetDefaultBrowserActivity.java [hunk 6]
```diff
@@ -74,4 +76,4 @@ public void onBackPressed() {}
     public boolean shouldStartGpuProcess() {
         return false;
     }
-}
+}
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/set_default_browser/SetDefaultBrowserActivity.java:79]: nit: new line

File: android/java/org/chromium/chrome/browser/set_default_browser/SetDefaultBrowserBottomSheetFragment.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+            }
+        }));
+    }
+
+    @Override
+    public void onDismiss(@NonNull DialogInterface dialog) {
+        super.onDismiss(dialog);
+        BraveSetDefaultBrowserUtils.isBottomSheetVisible = false;
+    }
+}
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/set_default_browser/SetDefaultBrowserBottomSheetFragment.java:127]: nit: new line

File: android/java/res/drawable/set_default_rounded_button_disabled.xml [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+     This Source Code Form is subject to the terms of the Mozilla Public
+     License, v. 2.0. If a copy of the MPL was not distributed with this file,
+     You can obtain one at http://mozilla.org/MPL/2.0/.
+-->
+
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle" android:padding="10dp">
+    <solid android:color="@color/set_default_bottom_sheet_button_disable_color"/>
+    <corners android:radius="50dp"/>
+</shape>
```
Comment (samartnik) [inline android/java/res/drawable/set_default_rounded_button_disabled.xml:12]: nit: new line

File: android/java/res/layout/fragment_set_default_browser_bottom_sheet.xml [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_horizontal"
+        android:button="@drawable/brave_checkbox"
+        android:layout_marginTop="20dp"
+        android:paddingStart="8dp"
+        android:text="@string/brave_dont_ask_again"
+        android:textSize="14sp"
+        android:textColor="@color/set_default_text_color"/>
+    
+</LinearLayout>
```
Comment (samartnik) [inline android/java/res/layout/fragment_set_default_browser_bottom_sheet.xml:94]: nit: new line

File: browser/ui/android/default_browser_promo/java/src/org/chromium/chrome/browser/ui/default_browser_promo/BraveDefaultBrowserPromoUtils.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * A utility class providing information regarding states of default browser.
+ */
+public class BraveDefaultBrowserPromoUtils {
+    
+    public static boolean prepareLaunchPromoIfNeeded(
+            Activity activity, WindowAndroid windowAndroid) {
+        return false;
+    }   
+}
```
Comment (samartnik) [inline browser/ui/android/default_browser_promo/java/src/org/chromium/chrome/browser/ui/default_browser_promo/BraveDefaultBrowserPromoUtils.java:19]: nit: new line

File: browser/ui/android/default_browser_promo/java/src/org/chromium/chrome/browser/ui/default_browser_promo/BraveDefaultBrowserPromoUtils.java [hunk 11]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
```
Comment (samartnik) [inline browser/ui/android/default_browser_promo/java/src/org/chromium/chrome/browser/ui/default_browser_promo/BraveDefaultBrowserPromoUtils.java:1]: Looks like it's Chromium's header, we have different in our java files.

File: browser/ui/android/strings/android_brave_strings.grd [hunk 12]
```diff
@@ -455,9 +455,12 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
       <message name="IDS_CONTINUE_TO_WALLET" desc="Text for continue to wallet button">
         Continue to wallet
       </message>
-      <message name="IDS_MENU_SET_DEFAULT_BROWSER" desc="Title for set default browser menu item">
+      <message name="IDS_SET_DEFAULT_BROWSER" desc="Title for set default browser menu item">
         Set as default browser
       </message>
+      <message name="IDS_MENU_SET_DEFAULT_BROWSER" desc="Title for set default browser menu item">
```
Comment (samartnik) [inline browser/ui/android/strings/android_brave_strings.grd:461]: It has same desc as previous strings, are these 2 the same?

File: build/android/bytecode/java/org/brave/bytecode/BraveDefaultBrowserPromoUtilsClassAdapter.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    static String sDefaultBrowserPromoUtilsClassName = "org/chromium/chrome/browser/ui/default_browser_promo/DefaultBrowserPromoUtils";
+
+    static String sBraveDefaultBrowserPromoUtilsClassName = "org/chromium/chrome/browser/ui/default_browser_promo/BraveDefaultBrowserPromoUtils";
+
+    public BraveDefaultBrowserPromoUtilsClassAdapter(ClassVisitor visitor) {
+        super(visitor);
+
+        changeMethodOwner(sDefaultBrowserPromoUtilsClassName, "prepareLaunchPromoIfNeeded", sBraveDefaultBrowserPromoUtilsClassName);
+    }
+}
```
Comment (samartnik) [inline build/android/bytecode/java/org/brave/bytecode/BraveDefaultBrowserPromoUtilsClassAdapter.java:20]: nit: new line

File: build/android/bytecode/java/org/brave/bytecode/BraveDefaultBrowserPromoUtilsClassAdapter.java [hunk 14]
```diff
@@ -0,0 +1,20 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
```
Comment (samartnik) [inline build/android/bytecode/java/org/brave/bytecode/BraveDefaultBrowserPromoUtilsClassAdapter.java:1]: 2022

File: build/android/bytecode/java/org/brave/bytecode/BraveDefaultBrowserPromoUtilsClassAdapter.java [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+
+public class BraveDefaultBrowserPromoUtilsClassAdapter extends BraveClassVisitor {
+    static String sDefaultBrowserPromoUtilsClassName = "org/chromium/chrome/browser/ui/default_browser_promo/DefaultBrowserPromoUtils";
+
+    static String sBraveDefaultBrowserPromoUtilsClassName = "org/chromium/chrome/browser/ui/default_browser_promo/BraveDefaultBrowserPromoUtils";
+
+    public BraveDefaultBrowserPromoUtilsClassAdapter(ClassVisitor visitor) {
+        super(visitor);
+
+        changeMethodOwner(sDefaultBrowserPromoUtilsClassName, "prepareLaunchPromoIfNeeded", sBraveDefaultBrowserPromoUtilsClassName);
```
Comment (samartnik) [inline build/android/bytecode/java/org/brave/bytecode/BraveDefaultBrowserPromoUtilsClassAdapter.java:18]: Please a test to `testMethodsExist` in `BytecodeTest.java` to make sure `prepareLaunchPromoIfNeeded` is still in `DefaultBrowserPromoUtils` in further Chromium versions.

**Review by samartnik — CHANGES_REQUESTED** [review]

**Comment by SergeyZhukovsky** [comment]
could you make sure the code runs on Android 6? Android 6 doesn't support all vector images especially with gradient, it crashes on them if doesn't support.

File: android/java/org/chromium/chrome/browser/notifications/retention/RetentionNotificationUtil.java [hunk 16]
```diff
@@ -19,10 +19,10 @@
 import org.chromium.base.IntentUtils;
 import org.chromium.base.Log;
 import org.chromium.chrome.R;
+import org.chromium.chrome.browser.app.BraveActivity;
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/notifications/retention/RetentionNotificationUtil.java:22]: I can't find where is used this import.

File: android/java/org/chromium/chrome/browser/set_default_browser/BraveSetDefaultBrowserUtils.java [hunk 17]
```diff
@@ -0,0 +1,234 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.chromium.chrome.browser.set_default_browser;
+
+import android.app.Activity;
+import android.app.role.RoleManager;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/set_default_browser/BraveSetDefaultBrowserUtils.java:14]: this `import android.content.SharedPreferences;` seems unused

File: android/java/org/chromium/chrome/browser/set_default_browser/SetDefaultBrowserBottomSheetFragment.java [hunk 18]
```diff
@@ -0,0 +1,127 @@
...
+ * Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.chromium.chrome.browser.set_default_browser;
+
+import android.content.DialogInterface;
+import android.content.res.Configuration;
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/set_default_browser/SetDefaultBrowserBottomSheetFragment.java:11]: these seems unused:
```
import android.content.res.Configuration;
import android.text.SpannableString;
import android.text.Spanned;
import android.text.style.ForegroundColorSpan;
import android.util.DisplayMetrics;
import android.view.WindowManager;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import androidx.appcompat.widget.AppCompatImageView;
import org.chromium.base.ContextUtils;
import org.chromium.chrome.browser.app.BraveActivity;
import org.chromium.chrome.browser.util.ConfigurationUtils;
import org.chromium.ui.base.DeviceFormFactor;
import java.util.Arrays;
import java.util.List;
```

File: android/java/res/drawable/set_default_rounded_button_disabled.xml [hunk 19]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) 2019 The Brave Authors. All rights reserved.
```
Comment (AlexeyBarabash) [inline android/java/res/drawable/set_default_rounded_button_disabled.xml:2]: 2022

File: android/java/res/values-night/brave_colors.xml [hunk 20]
```diff
@@ -28,6 +28,10 @@
     <!-- Text colors -->
     <color name="brave_theme_hint_text">#84889C</color>
 
+    <!-- Set Default Browser colors -->
+    <color name="set_default_bottom_sheet_bg_color">#1E2029</color>
```
Comment (AlexeyBarabash) [inline android/java/res/values-night/brave_colors.xml:32]: :+1: for putting colors here

File: android/brave_java_sources.gni (hunk 1 above)
Comment (tapanmodh) [reply android/brave_java_sources.gni]: Fixed it. It should be ok now

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java (hunk 2 above)
Comment (tapanmodh) [reply android/java/org/chromium/chrome/browser/app/BraveActivity.java]: removed it

File: android/java/org/chromium/chrome/browser/notifications/retention/RetentionNotificationUtil.java (hunk 3 above)
Comment (tapanmodh) [reply android/java/org/chromium/chrome/browser/notifications/retention/RetentionNotificationUtil.java]: sure

File: android/java/org/chromium/chrome/browser/set_default_browser/BraveSetDefaultBrowserUtils.java (hunk 4 above)
Comment (tapanmodh) [reply android/java/org/chromium/chrome/browser/set_default_browser/BraveSetDefaultBrowserUtils.java]: added new line in all the files

File: browser/ui/android/default_browser_promo/java/src/org/chromium/chrome/browser/ui/default_browser_promo/BraveDefaultBrowserPromoUtils.java (hunk 11 above)
Comment (tapanmodh) [reply browser/ui/android/default_browser_promo/java/src/org/chromium/chrome/browser/ui/default_browser_promo/BraveDefaultBrowserPromoUtils.java]: changed it our copyright text

File: browser/ui/android/strings/android_brave_strings.grd (hunk 12 above)
Comment (tapanmodh) [reply browser/ui/android/strings/android_brave_strings.grd:461]: changed desc

File: build/android/bytecode/java/org/brave/bytecode/BraveDefaultBrowserPromoUtilsClassAdapter.java (hunk 14 above)
Comment (tapanmodh) [reply build/android/bytecode/java/org/brave/bytecode/BraveDefaultBrowserPromoUtilsClassAdapter.java]: done

File: build/android/bytecode/java/org/brave/bytecode/BraveDefaultBrowserPromoUtilsClassAdapter.java (hunk 15 above)
Comment (tapanmodh) [reply build/android/bytecode/java/org/brave/bytecode/BraveDefaultBrowserPromoUtilsClassAdapter.java]: done. can you please check if it is proper?

File: android/java/org/chromium/chrome/browser/set_default_browser/BraveSetDefaultBrowserUtils.java (hunk 17 above)
Comment (tapanmodh) [reply android/java/org/chromium/chrome/browser/set_default_browser/BraveSetDefaultBrowserUtils.java]: removed

File: android/java/org/chromium/chrome/browser/set_default_browser/SetDefaultBrowserBottomSheetFragment.java (hunk 18 above)
Comment (tapanmodh) [reply android/java/org/chromium/chrome/browser/set_default_browser/SetDefaultBrowserBottomSheetFragment.java]: ohh! removed all of them. Thanks :)

File: android/java/res/drawable/set_default_rounded_button_disabled.xml (hunk 19 above)
Comment (tapanmodh) [reply android/java/res/drawable/set_default_rounded_button_disabled.xml]: My bad. done

**Review by samartnik — APPROVED** [review]
lgtm

**Review by AlexeyBarabash — APPROVED** [review]
++

**Comment by tapanmodh** [comment]
> could you make sure the code runs on Android 6? Android 6 doesn't support all vector images especially with gradient, it crashes on them if doesn't support.

Yes, It's looks fine on Android 6.

**Comment by SergeyZhukovsky** [comment]
> > could you make sure the code runs on Android 6? Android 6 doesn't support all vector images especially with gradient, it crashes on them if doesn't support.
> 
> Yes, It's looks fine on Android 6.

perfect, thanks for checking it!

**Review by deeppandya — APPROVED** [review]
LGTM

**Pull request merged**